pylint/checkers/strings.py

```diff
@@ -234,6 +239,22 @@ class StringFormatChecker(BaseChecker):
     msgs = MSGS
 
     @check_messages(*(MSGS.keys()))
+    def visit_list(self, node):
+        self.check_for_concatenated_strings(node, 'list')
+
+    @check_messages(*(MSGS.keys()))
+    def visit_tuple(self, node):
+        self.check_for_concatenated_strings(node, 'tuple')
+
+    def check_for_concatenated_strings(self, iterable_node, iterable_type):
+        # This check relies on the asumption that StringConstantChecker is run beforehand
+        for elt in iterable_node.elts:
+            if isinstance(elt, astroid.node_classes.Const) and elt.pytype() == 'builtins.str':
+                if StringConstantChecker.string_tokens[(elt.lineno, elt.col_offset)] != elt.value:
+                    self.add_message('str-literals-concatenation-and-comma',
+                                     node=iterable_node, args=(iterable_type,))
+
+    @check_messages(*(set(MSGS.keys()) ))
```

Comment:
creating set from dict keys is redundant.

Reply:
Code was already there :)
But ok, boy scout rule applies ;)
![](https://chezsoi.org/lucas/wwcb/photos/BoyScoutRule.jpg)